src/compute/src/row_spine.rs

```diff
@@ -208,15 +207,15 @@ mod container {
         fn index(&self, index: usize) -> &[u8] {
             let lower = self.offsets.index(index);
             let upper = self.offsets.index(index + 1);
-            &self.storage[lower..upper]
+            &self.storage[*lower..*upper]
         }
         fn len(&self) -> usize {
             self.offsets.len() - 1
         }
 
         fn with_capacities(item_cap: usize, byte_cap: usize) -> Self {
             // TODO: be wary of `byte_cap` greater than 2^32.
-            let mut offsets = OffsetList::with_capacity(item_cap + 1);
+            let mut offsets = crate::row_spine::OffsetOptimized::with_capacity(item_cap + 1);
```

Comment:
The `OffsetOptimized::with_capacity` call ignores its argument. Perhaps we should think about that to be sure we like that behavior.